src/lvm.cpp

```diff
@@ -0,0 +1,451 @@
+          "Legacy\"";
+
+    int exitCode = cloyster::runCommand(checkUEFICommand, output, false);
+
+    if (exitCode == 0) {
+        auto it = std::find(output.begin(), output.end(), "UEFI");
+        if (it != output.end()) {
+            LOG_TRACE("LVM: System boot mode: UEFI\n");
+        } else {
+            LOG_WARN("LVM: System is not booted in UEFI mode.");
+        }
+    } else {
+        LOG_WARN("LVM: Failed to check system boot mode.");
+    }
+}
+
+void LVM::checkLVMEnabled()
+{
+    std::list<std::string> output;
+    const std::string checkLVMCommand = "vgs --noheadings";
+    int exitCode = cloyster::runCommand(checkLVMCommand, output, false);
+
+    if (exitCode == 0) {
+        if (!output.empty()) {
+            LOG_INFO("LVM is enabled.\n");
+        } else {
+            throw std::runtime_error("LVM ERROR: LVM is not enabled.");
+        }
+    } else {
+        throw std::runtime_error(
+            "LVM ERROR: Failed to check if LVM is enabled.");
+    }
+}
+
+void LVM::checkThinProvisioning()
+{
+    std::list<std::string> output;
+    const std::string checkThinCommand = "lvs -o+lv_layout --noheadings";
+    int exitCode = cloyster::runCommand(checkThinCommand, output, false);
+
+    if (exitCode == 0) {
+        auto it = std::find_if(
+            output.begin(), output.end(), [](const std::string& line) {
+                return line.find("thin") != std::string::npos;
+            });
+
+        if (it != output.end()) {
+            LOG_TRACE("LVM Thin provisioning is enabled.\n");
+        } else {
+            LOG_WARN("LVM Thin provisioning is disabled.");
+        }
+    } else {
+        LOG_WARN("Failed to check thin provisioning status.");
+    }
+}
+
+void LVM::checkEnoughDiskSpaceAvailable()
+{
+    std::list<std::string> output;
+    const std::string checkDiskSpaceCommand
+        = "vgs --noheadings -o vg_name,vg_size,vg_free --units G";
+
+    int exitCode = cloyster::runCommand(checkDiskSpaceCommand, output, false);
+
+    if (exitCode == 0 && !output.empty()) {
+        try {
+            bool allVGsHaveEnoughSpace = true;
+
+            for (const std::string& line : output) {
+                LOG_DEBUG("LVM: Check disks space - Raw output line: {}", line);
+
+                // Remove leading/trailing whitespaces
+                std::string cleanLine = line;
+                cleanLine.erase(cleanLine.begin(),
+                    std::find_if(cleanLine.begin(), cleanLine.end(),
+                        [](unsigned char ch) { return !std::isspace(ch); }));
+                cleanLine.erase(
+                    std::find_if(cleanLine.rbegin(), cleanLine.rend(),
+                        [](unsigned char ch) { return !std::isspace(ch); })
+                        .base(),
+                    cleanLine.end());
+
+                if (cleanLine.empty()) {
+                    LOG_DEBUG("LVM: Skipping empty or malformed line.");
```

Comment:
If the statement dispose away lines, it should at least say what was thrown away. Perhaps with `LOG_TRACE`?

Reply:
It shows the raw line if running with LOG_DEBUG
`LOG_DEBUG("LVM: Check disks space - Raw output line: {}", line);`